data_model/src/asset.rs

```diff
@@ -263,122 +294,110 @@ pub struct DefinitionId {
 )]
 pub struct Id {
     /// Entity Identification.
-    pub definition_id: DefinitionId,
+    pub definition_id: <AssetDefinition as Identifiable>::Id,
     /// Account Identification.
-    pub account_id: AccountId,
+    pub account_id: <Account as Identifiable>::Id,
 }
```

Comment:
At the moment I don't have much incentive from the FFI to make these fields private.
Would you like to make them private? It would bear no significance for FFI performance or API

Reply:
From a generalist API perspective, we should make things private as much as possible

Reply:
I'm not going to do it in this PR now, but thanks for stating your opinion. I'll have it in mind